### PR TITLE
FLUID-6497: Can now set any cookie attribute.

### PR DESCRIPTION
**Pull request by jobara**
Updates defaults to set samesite attribute.

https://issues.fluidproject.org/browse/FLUID-6497

### CODE REVIEW

File: src/framework/preferences/js/Store.js [hunk 1]
```diff
@@ -128,13 +131,14 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
...
-        if (options.expires) {
-            cookieStr += "; expires=" + options.expires;
-        }
-
-        if (options.path) {
-            cookieStr += "; path=" + options.path;
-        }
+        fluid.each(options, function (value, setting) {
+            // skip name because it was already set as the first portion of the cookieStr
+            if (setting !== "name" && setting !== "secure") {
```
Comment (cindyli) [inline src/framework/preferences/js/Store.js:136]: `HttpOnly` is another no-value cookie setting, similar as `secure`. See [this documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Syntax). It's good to handle it too.

Comment (jobara) [reply src/framework/preferences/js/Store.js:136]: The cookie store is meant to be used in the browser, as it uses [document.cookie](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie). If you look at the JavaScript section [Creating Cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Creating_cookies) it says:

> Cookies created via JavaScript cannot include the HttpOnly flag.

File: src/framework/preferences/js/Store.js [hunk 2]
```diff
@@ -128,13 +131,14 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
...
         var cookieStr = cookieName + "=" + data;
 
-        if (options.expires) {
-            cookieStr += "; expires=" + options.expires;
-        }
-
-        if (options.path) {
-            cookieStr += "; path=" + options.path;
-        }
+        fluid.each(options, function (value, setting) {
```
Comment (amb26) [inline src/framework/preferences/js/Store.js:134]: While we are here, could we take the opportunity to improve the docs of fluid.prefs.cookieStore.assembleCookie - e.g. more accurate type for "options" - perhaps Object.<String, String>, or perhaps with a typedef given this argument type appears in more than one function, and a list of at least the settings that we know we support - perhaps with a link to the MDN documentation that you quoted.
Also perhaps fix the minor anomaly that the argument is called "payload" in writeCookie but "data" in assembleCookie. It looks like the argument type in writeCookie must be wrong

Comment (amb26) [reply src/framework/preferences/js/Store.js:134]: Other comment issues include the use of "SettingsStore" where this has now been migrated over to DataSources, and the use of "Subcomponent" is anachronistic

**Comment by jobara** [comment]
@cindyli and @amb26 this is ready for more review.

File: tests/framework-tests/preferences/js/StoreTests.js [hunk 3]
```diff
@@ -48,31 +48,56 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
...
+            },
+            expected: "cookieName=cookieValue"
         }, {
-            testName: "no path or expiry date set",
+            testName: "only name set",
             cookieName: "cookieName",
             payload: "cookieValue",
             expected: "cookieName=cookieValue"
+        }, {
+            testName: "name attribute in cookie options",
```
Comment (cindyli) [inline tests/framework-tests/preferences/js/StoreTests.js:94]: Probably add a test case that the name attribute in cookie options is applied when `cookieName` is not provided.

Comment (jobara) [reply tests/framework-tests/preferences/js/StoreTests.js:93]: `fluid.prefs.cookieStore.assembleCookie` doesn't handle the name calculations. That is handled by `fluid.prefs.cookieStore.writeCookie`. There already tests in place against that function. However, I've done a bit of refactoring to make the test reporting a bit clearer, and simplified some other parts of the test code.

**Pull request merged**